src/User-repo.js

```diff
@@ -1,84 +1,12 @@
-    return totalStepGoal / this.users.length;
-  };
-  makeSortedUserArray(id, dataSet) {
-    let selectedID = this.getDataFromUserID(id, dataSet)
-    let sortedByDate = selectedID.sort((a, b) => new Date(b.date) - new Date(a.date));
-    return sortedByDate;
-  }
-  getToday(id, dataSet) {
-    return this.makeSortedUserArray(id, dataSet)[0].date;
-  };
-  getFirstWeek(id, dataSet) {
-    return this.makeSortedUserArray(id, dataSet).slice(0, 7);
-  };
-  getWeekFromDate(date, id, dataSet) {
-    let dateIndex = this.makeSortedUserArray(id, dataSet)
-      .indexOf(this.makeSortedUserArray(id, dataSet)
-      .find((sortedItem) => (sortedItem.date === date)));
-    return this.makeSortedUserArray(id, dataSet).slice(dateIndex, dateIndex + 7);
-  };
-  chooseWeekDataForAllUsers(dataSet, date) {
-    return dataSet.filter(function(dataItem) {
-      return (new Date(date)).setDate((new Date(date))
-        .getDate() - 7) <= new Date(dataItem.date) && new Date(dataItem.date) <= new Date(date)
-    })
-  };
-  chooseDayDataForAllUsers(dataSet, date) {
-    return dataSet.filter(function(dataItem) {
-      return dataItem.date === date
-    });
-  }
-  isolateUsernameAndRelevantData(dataSet, date, relevantData, listFromMethod) {
-    return listFromMethod.reduce(function(objectSoFar, dataItem) {
-      if (!objectSoFar[dataItem.userID]) {
-        objectSoFar[dataItem.userID] = [dataItem[relevantData]]
-      } else {
-        objectSoFar[dataItem.userID].push(dataItem[relevantData])
-      }
-      return objectSoFar;
-    }, {});
-  }
-  rankUserIDsbyRelevantDataValue(dataSet, date, relevantData, listFromMethod) {
-    let sortedObjectKeys = this.isolateUsernameAndRelevantData(dataSet, date, relevantData, listFromMethod)
-    return Object.keys(sortedObjectKeys).sort(function(b, a) {
-      return (sortedObjectKeys[a].reduce(function(sumSoFar, sleepQualityValue) {
-        sumSoFar += sleepQualityValue
-        return sumSoFar;
-      }, 0) / sortedObjectKeys[a].length) - (sortedObjectKeys[b].reduce(function(sumSoFar, sleepQualityValue) {
-        sumSoFar += sleepQualityValue
-        return sumSoFar;
-      }, 0) / sortedObjectKeys[b].length)
-    });
-  }
-  combineRankedUserIDsAndAveragedData(dataSet, date, relevantData, listFromMethod) {
-    let sortedObjectKeys = this.isolateUsernameAndRelevantData(dataSet, date, relevantData, listFromMethod)
-    let rankedUsersAndAverages = this.rankUserIDsbyRelevantDataValue(dataSet, date, relevantData, listFromMethod)
-    return rankedUsersAndAverages.map(function(rankedUser) {
-      rankedUser = {
-        [rankedUser]: sortedObjectKeys[rankedUser].reduce(
-          function(sumSoFar, sleepQualityValue) {
-            sumSoFar += sleepQualityValue
-            return sumSoFar;
-          }, 0) / sortedObjectKeys[rankedUser].length
-      };
-      return rankedUser;
-    });
+    return this.users.map(user => user.dailyStepGoal).reduce((total, goal) => total += goal, 0) / this.users.length;
```

Comment:
I am not currently seeing where the methods have been relocated.  Maybe a stand-up is in order to run through new features and function outsourcing.